### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by AnjaLumholtz**
Potential fix

### CODE REVIEW

File: JANGAN/ExperimentQueue/TestRunConfig.ini [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Do not change values in here for local changes
```
Comment (jamadaha) [inline JANGAN/ExperimentQueue/TestRunConfig.ini:1]: Is this file intended to be uploaded?

File: JANGAN/ExperimentQueue/TestRun.py [hunk 2]
```diff
@@ -0,0 +1 @@
+# Change functions and methods, to fit the goal of the experiment
```
Comment (jamadaha) [inline JANGAN/ExperimentQueue/TestRun.py:1]: Is this file intended to be uploaded?

File: JANGAN/ExperimentQueue/Test2RunConfig.ini [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# Do not change values in here for local changes
```
Comment (jamadaha) [inline JANGAN/ExperimentQueue/Test2RunConfig.ini:1]: Is this file intended to be uploaded?

File: JANGAN/ExperimentQueue/Test2Run.py [hunk 4]
```diff
@@ -0,0 +1 @@
+# Change functions and methods, to fit the goal of the experiment
```
Comment (jamadaha) [inline JANGAN/ExperimentQueue/Test2Run.py:1]: Is this file intended to be uploaded?

File: JANGAN/ExperimentQueueConfig.ini [hunk 5]
```diff
@@ -17,4 +17,28 @@ AmountOfTimesToRun = 1
...
+MakeCGANDataset = True
+TrainCGAN = True
+ProduceCGANLetters = True
+
+[TEST3RUN]
+ModuleName = "ExperimentQueue.Test2Run"
+AmountOfTimesToRun = 1
+ConfigFile = "ExperimentQueue/Test2RunConfig.ini"
+MakeCGANDataset = True
+TrainCGAN = True
```
Comment (jamadaha) [inline JANGAN/ExperimentQueueConfig.ini:43]: Are these meaningful?

File: JANGAN/CheckMemoryLeak/__init__.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+class CheckMemoryLeak():
+    TraceMallocPath = ""
+    TraceMallocFile = ""
+
+    Snapshot = {}
+    SnapshotKeys = []
+    Top_stats = any
+
+    def ConfigurePath(self):
+        self.TracemallocPath = '../../Data/TraceMalloc'
```
Comment (jamadaha) [inline JANGAN/CheckMemoryLeak/__init__.py:15]: Should this be a config option?

File: JANGAN/CheckMemoryLeak/__init__.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+        if not os.path.isdir(self.TracemallocPath):
+            os.makedirs(self.TracemallocPath)
+
+        named_tuple = time.localtime() # get struct_time
+        time_string = time.strftime("%m-%d-%Y, %H:%M:%S", named_tuple)
+
+        self.TracemallocFile = '/' + time_string + ' tracemalloc.txt'
```
Comment (jamadaha) [inline JANGAN/CheckMemoryLeak/__init__.py:23]: Could the name be a config option?

**Comment by jamadaha** [comment]
Tested, seems fine, except for this:
image
Which occurred after running 4 experiments in 1 queue.

File: JANGAN/CheckMemoryLeak/__init__.py (hunk 6 above)
Comment (HGinnerup) [reply JANGAN/CheckMemoryLeak/__init__.py:15]: I'd think so, if we intend to keep the profiler in the code

File: JANGAN/CheckMemoryLeak/__init__.py (hunk 7 above)
Comment (HGinnerup) [reply JANGAN/CheckMemoryLeak/__init__.py:23]: Putting it in the config would also remove the need for `time_string` as the config now supports `../../Data/{TIMESTAMP}/TraceMalloc/tracemalloc.txt`

**Review by HGinnerup — CHANGES_REQUESTED** [review]
TraceMalloc looks like something that will add overhead to the whole program.

If it is to remain in the program, maybe make it optional via the config?

**Comment by AnjaLumholtz** [comment]
> Tested, seems fine, except for this: image Which occurred after running 4 experiments in 1 queue.

Fixed

**Review by jamadaha — APPROVED** [review]

File: JANGAN/GlobalConfig.ini [hunk 8]
```diff
@@ -0,0 +1,8 @@
+[TRACEMALLOC]
+
+# Check whether to run tracemalloc
+RunTraceMalloc = True
```
Comment (HGinnerup) [inline JANGAN/GlobalConfig.ini:4]: Don't enable TraceMalloc on main

**Review by HGinnerup — CHANGES_REQUESTED** [review]

**Review by HGinnerup — APPROVED** [review]

**Pull request merged**